protocol/v2/ssv/validator/committee.go

```diff
@@ -87,6 +90,9 @@ func (c *Committee) RemoveShare(validatorIndex phase0.ValidatorIndex) {
 	if share, exist := c.Shares[validatorIndex]; exist {
 		c.dutyGuard.StopValidator(share.ValidatorPubKey)
 		delete(c.Shares, validatorIndex)
+		if len(c.Shares) == 0 {
+			c.stop()
```

Comment:
@y0sher now that we are able to stop the committee should we be checking its state before doing actions like `AddShare`, `ProcessMessage` etc?

Reply:
Perhaps it's better (more straightforward) to just call `Committee.cancel` once we determined that committee can stop (shares dropped to 0), presumably this should be enough to finish all the work `Committee` was doing (while it's `controller`'s job to make sure no new tasks will be assigned to such `Committee` - which it does handle by atomically removing it from `controller.validatorsMap`),

and, from my reading of this code, unless we do something with stopped `Committee` (have some business logic for it in stopped state, like restarting it) there is no reason to even store this `stopped atomic.Bool` (and have corresponding methods `stop` and `Stopped`) - you could simply remove these and adjust the way `UpdateCommitteeAtomic` works (so that instead of checking `Stopped()` func `mutate` will return a `bool` to signal that committee was stopped by this mutation).